src/WebJobs.Script.WebHost/FunctionsSyncService.cs

```diff
@@ -69,12 +75,23 @@ private async void OnSyncTimerTick(object state)
         {
             try
             {
-                var cancellationToken = (CancellationToken)state;
-
-                if (!cancellationToken.IsCancellationRequested && ShouldSyncTriggers)
+                // Shutdown if RunFromZipFailed
+                string runFromPackageFailedPath = Path.Combine(_applicationHostOptions.CurrentValue.ScriptPath, ScriptConstants.RunFromPackageFailedFileName);
```

Comment:
This issue isn't specific to SyncTriggers - e.g. Any admin operations that return file info will be affected if the file system isn't initialized properly. Also, the FunctionsSyncService is only conditionally registered, whereas we want this check to always be performed.

So I think you should move this check to WebJobsScriptHostService